errbot/backends/irc.py

```diff
@@ -481,8 +495,56 @@ def on_join(self, connection, event):
         # We need to wait to endofnames message.
         room_name = event.target
         with self._recently_joined_lock:
+            t = IRCRoom(room_name, self.bot)
             self._recently_joined_to.add(room_name)
 
+    def on_currenttopic(self, connection, event):
+        """
+            When you Join a room with a topic set this event fires up to
+            with the topic information.
+            If the room that you join don't have a topic set, nothing happens.
+            :param:
+                connection: Is an 'irc.client.ServerConnection' object
+
+            :param:
+                event: Is an 'irc.client.Event' object
+                the event.arguments[1] contains the topic of the room.
+                the event.arguments[0] contains the room name
+        """
+        room_name = event.arguments[0]
+        self.bot._rooms[room_name]._topic = event.arguments[1]
+
+    def on_topic(self, connection, event):
+        """
+            On response to the /TOPIC command if the room have a topic.
+            If the room don't have a topic the event fired is on_notopic
+            :param:
+                connection: Is an 'irc.client.ServerConnection' object
+
+            :param:
+                event: Is an 'irc.client.Event' object
+                the event.target contains the room name.
+                the event.arguments[0] contains the topic name
+        """
+        room_name = event.target
+        self.bot._rooms[room_name]._topic = event.arguments[0]
```

Comment:
Race condition on `self.bot._rooms` here and same comment as above about directly modifying `_topic`.